packages/block-editor/src/text-input/sidebar.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * External dependencies
+ */
+import {
+	// eslint-disable-next-line
+	__experimentalToggleGroupControl as ToggleGroupControl,
+	// eslint-disable-next-line
+	__experimentalToggleGroupControlOption as ToggleGroupControlOption,
+	PanelBody,
+	TextControl,
+	ToggleControl,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { InspectorControls } from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import ColorSettings from '../components/color-settings';
+
+export default ( { attributes, setAttributes } ) => {
+	const widthOptions = [ 25, 50, 75, 100 ];
+
+	const handleChangeAttribute = ( key ) => ( value ) =>
+		setAttributes( {
+			[ key ]: value,
+		} );
+
+	return (
+		<InspectorControls>
+			<PanelBody
+				title={ __( 'Field Settings', 'blocks' ) }
+				initialOpen={ true }
+			>
+				<ToggleControl
+					label={ __( 'The field is required' ) }
+					checked={ attributes.mandatory }
+					onChange={ handleChangeAttribute( 'mandatory' ) }
+				/>
+				<TextControl
+					label="Input field height"
+					type="number"
+					value={ attributes.inputHeight }
+					onChange={ handleChangeAttribute( 'inputHeight' ) }
```

Comment:
I believe there might be issue with this one.  
`handleChangeAttribute` saves the attribute as is, which means it'd attempt saving `inputHeight` as a string which will work in the editor but the changes will not actually get persisted when you publish or refresh the page.  
To solve this we need another helper and run `parseInt` on the value before it's passed to `setAttribtues`.

Reply:
Nice catch! Thanks for clarifying.